Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -215,29 +215,36 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
                     mc_corrections::WeightType::BTag);
 
 
-                    static const std::vector<UncertaintySource> btag_sources = { UncertaintySource::btag_lf,
-                        UncertaintySource::btag_hf, UncertaintySource::btag_hfstats1, UncertaintySource::btag_hfstats2,
-                        UncertaintySource::btag_lfstats1, UncertaintySource::btag_lfstats2,
-                        UncertaintySource::btag_cferr1, UncertaintySource::btag_cferr2,
-                        UncertaintySource::JetFull_Total, UncertaintySource::JetReduced_Total
+                    static const std::vector<UncertaintySource> btag_sources = {
+                        UncertaintySource::btag_lf, UncertaintySource::btag_hf, UncertaintySource::btag_hfstats1,
+                        UncertaintySource::btag_hfstats2, UncertaintySource::btag_lfstats1,
+                        UncertaintySource::btag_lfstats2, UncertaintySource::btag_cferr1, UncertaintySource::btag_cferr2
                      };
 
 
                 for(const bool iter_fit : {false, true}) {
                     for(const auto wp : btag_wps){
                         const std::pair<DiscriminatorWP, bool> key(wp, iter_fit);
-                        btag_weights[key][UncertaintyScale::Central] = static_cast<float>(
-                                btag_weight_provider->Get(*event, wp, iter_fit, unc_source, unc_scale));
+                        btag_weights.weights[wp][UncertaintyScale::Central] = static_cast<float>(
+                                btag_weight_provider->Get(*event, wp, iter_fit, unc_source, unc_scale, false));
                         if(unc_source == UncertaintySource::None && !iter_fit) {
                             for(const auto scale : {UncertaintyScale::Up, UncertaintyScale::Down})
-                                btag_weights[key][scale] = static_cast<float>(btag_weight_provider->Get(*event, wp,
-                                    iter_fit, UncertaintySource::Eff_b, scale));
+                                btag_weights.weights[wp][scale] = static_cast<float>(btag_weight_provider->Get(*event,
+                                    wp, iter_fit, UncertaintySource::Eff_b, scale, false));
                         }
                         else if(unc_source == UncertaintySource::None && iter_fit) {
                             for(const auto scale : {UncertaintyScale::Up, UncertaintyScale::Down}) {
-                                for(UncertaintySource unc : btag_sources) {
-                                    btag_weights[key][scale] = static_cast<float>(
-                                        btag_weight_provider->Get(*event, wp, iter_fit, unc, scale));
+                                for(const bool apply_JES : {false, true}) {
```

Comment:
it is still inside `if(unc_source == UncertaintySource::None)`, while it shouldn't be